src/main/java/com/commercetools/project/sync/Syncer.java

```diff
@@ -257,6 +258,9 @@ public B getSync() {
     return sync;
   }
 
+  @Nonnull
+  protected abstract Logger getLoggerInstance();
```

Comment:
I would not expose stuff solely for testing purposes as it decreases readability and introduces confusing and production un-relevant code noise. For example if you would want to mock it then you could have passed it with a class constructor but to be honest I doubt we need to assert against what we log but rather more reliable outcome of the process like returned stats object or data in commercetools.

Reply:
This is not added only for test purposes, Syncer is using a decorator pattern for the sync method and calling the abstract calls of the subclasses and some logs such as starting sync are part of the generic Syncer superclass, this change is added to put the right canonical class name into loggerName. Another option would be adding loggings as abstract calls so every Syncer sub class adds their logs by themself thus creates more boilerplate, that's why I choose this approach. Let me know if you are seeing a different approach so I could adapt.

Reply:
Maybe just replace [here](https://github.com/commercetools/commercetools-project-sync/pull/264/files#diff-c7b5dd4b409fbf4e3752df9e04b325a13475e75913e6ea6b06712b0aba78723cL570):

`private static final Logger LOGGER = LoggerFactory.getLogger(Syncer.class);`
with 
`protected final Logger log = LoggerFactory.getLogger(getClass());`

Note that with this approach instance is not always the same so static modifier is not beneficial. Also, I would replace long upper case variable name `LOGGER` to just `log` as it is a not a constant.

Reply:
This is logical, I also tried that, but in the context of the logger, it is not always instance level, for instance, `of` methods of Syncer subclasses are static factory methods, which injects the logger into the sync callbacks. 
Also not seeing any issue in terms of mocking the logger mostly it's not mocking but changing the sl4j implementation of the LogEvents and directing them to another implementation that stores logs into memory, for that we are using https://projects.lidalia.org.uk/slf4j-test/